content/blog/2024-09-09-fips-validation-for-almalinux-update.md

```diff
@@ -0,0 +1,54 @@
+
+When it comes to securing sensitive data, cryptographic standards play a crucial role. One of the most influential standards in this realm is **FIPS**, or **F**ederal **I**nformation **P**rocessing **S**tandards.
+
+FIPS 140-3 is not just a Federal benchmark, it's recognized and respected worldwide, for instance, the European Union's [NIS 2 Directive](https://eur-lex.europa.eu/eli/dir/2022/2555) which focuses on network and information security mandates the use of *up-to-date encryption and cryptography leveraging established standards.* FIPS standards undergo extensive evaluations to ensure that cryptographic algorithms and implementations meet the highest security requirements, making it the gold standard for secure cryptographic practices across the globe.
+
+Anyone who uses [AlmaLinux 9.2](https://almalinux.org/blog/almalinux-92-now-available/) can achieve FIPS compliance for FREE, the only thing you need to do is download the packages from the TuxCare repo [(docs)](https://docs.tuxcare.com/enterprise-support-for-almalinux/fips) and enable FIPS mode:
+
+```bash
+dnf -y install https://repo.tuxcare.com/fips/tuxcare-fips-release-latest-9.noarch.rpm
+dnf -y install openssl-3.0.7-20.el9_2.tuxcare.1 kernel-5.14.0-284.11.1.el9_2.tuxcare.5
+fips-mode-setup --enable
+reboot
+```
+
+This allows people with less knowledge about cryptography to configure a strong baseline for implementing secure systems.
+
+### Testing
+
+To test this, we can setup an HTTP server, for example [Apache](https://httpd.apache.org) - if you're interested, we have a comprehensive guide on [setting up a LAMP server](https://wiki.almalinux.org/series/LAMP-server.html) on AlmaLinux, but for now we can just run:
+
+```bash
+dnf -y install httpd mod_ssl
+systemctl start httpd
+```
+
+If you were to scan the server using nmap, you would see something like this:
+
+![nmap scan in fips mode](/blog-images/2024/nmap-fips.png)
+
+For reference, with FIPS mode disabled you get the weaker TLSv1.2 and non-compliant ciphersuites using Edwards curves, SHA1 hashes and ChaCha20-Poly1305 ciphers:
+
+![nmap scan in non-fips mode](/blog-images/2024/nmap-nonfips.png)
+
+Another way to check your ciphers strength is to look them up on the [Ciphersuite Info](https://ciphersuite.info) website or using the [sslscan](https://github.com/rbsec/sslscan) tool.
+
+### Whats next?
+
+Simon John, the Security Certification Manager at [CloudLinux](https://cloudlinux.com/) has let me know that after AlmaLinux 9.5 gets released he is going to start working on the 9.6 validation, which hopefully will materialize faster than 9.2 due to improvements in the CMVP process aimed at reducing the backlog.
```

Comment:
Can you work with Simon to also find somewhere to send folks if they are excited about security. For example, for the [CIS benchmark blog post](https://almalinux.org/blog/2023-11-28-cis-benchmarks-update/), we suggested they go participate in writing those. Maybe something similar here? or just send them to the security room chat.

Reply:
Yeah but I am not familiar with anything of sorts, I'll ask Simon.

Reply:
Wow that's a difficult one, FIPS isn't really something you can volunteer to help with.

There's a good overview of the CMVP process and docs here (bit boring!): https://csrc.nist.gov/Projects/cryptographic-module-validation-program/fips-140-3-standards

atsec (our labs) have a few good FIPS webinars here: https://www.atsec.com/category/fips-140-3/

or more generically there's https://nmap.org/ home of nmap, bugtraq and the full disclosure mailling list, they have a list of security scanner tools and books etc.

and erm, there's always the tuxcare blog: https://tuxcare.com/blog/

Reply:
bringing people to the chat might help with the "not busy" part, but totally get what you're saying. Maybe a call out to propose blog posts? "If you're doing something cool in the security space that involves AlmaLinux at all and want to write a blog post, join chat.almalinux.org and reach out in ~security!"

Reply:
So are we adding it to the blog or no? I'm confused 😂